crates/bevy_ecs/src/system/query.rs

```diff
@@ -640,6 +679,99 @@ where
         }
     }
 
+    /// Returns the read-only query results for the provided Array of [`Entity`]s.
+    ///
+    /// These values follow the order of your input Array (if anyy).
+    /// In case of a nonexisting entity or mismatched component,
+    /// a [`QueryEntityError`] is returned instead.
+    ///
+    /// If you need to verify the identity of each item returned,
+    /// add [`Entity`] to your [`Query`].
+    ///
+    /// # Example
+    /// ```rust
+    /// # use bevy_ecs::prelude::*;
+    /// #[derive(Component, PartialEq, Debug)]
+    /// struct A(u64);
+    ///
+    /// let mut world = World::new();
+    /// let entity_1 = world.spawn().insert(A(1)).id();
+    /// let entity_2 = world.spawn().insert(A(2)).id();
+    /// let entity_3 = world.spawn().insert(A(3)).id();
+    ///
+    /// let query_state = world.query::<&A>();
+    /// let a_query = Query::from_state(&mut world, &query_state);
+    /// let mut a_iterator = a_query.get_multiple([entity_3, entity_2, entity_1]).map(|i|i.unwrap());
+    /// assert_eq!(*a_iterator.next().unwrap(), A(3));
+    /// assert_eq!(*a_iterator.next().unwrap(), A(2));
+    /// assert_eq!(*a_iterator.next().unwrap(), A(1));
+    /// ```
+    #[inline]
+    pub fn get_multiple<const N: usize>(
+        &'s self,
+        entities: [Entity; N],
+    ) -> impl Iterator<Item = Result<<Q::ReadOnlyFetch as Fetch>::Item, QueryEntityError>> {
```

Comment:
```suggestion
        &self,
        entities: [Entity; N],
    ) -> impl Iterator<Item = Result<<Q::ReadOnlyFetch as Fetch<'_, 's>>::Item, QueryEntityError>> {
```
Though to be honest I'm not sure there is a need for this API given the impl is so trivial :woman_shrugging: